Yoast/wordpress-seo
# Filter out leading zeros in How-To Block time input fields

## Summary

This PR can be summarized in the following changelog entry:

* Leading 0s in the input fields for time in the How-To block are removed automatically. 

## Relevant technical choices:

* Added id's to the input fields for Hours and Minutes
* Set maximum duration in minutes to 59 (from 60).

## Test instructions

This PR can be tested by following these steps:

* Checkout this branch and build.
* Make a new post (with Gutenberg enabled), add a How-To block by clicking the "+" and scrolling down to "Structured data blocks" and selecting the "How-To" block.
* Add a time in hours and minutes, and make sure that when you type leading 0s (for example "00002" hours) the 0s are removed. So it should become "2". Trailing 0s (for example 20000) should work as usual. 
* The cursor will jump to the back (right) of the input field when a leading 0 is removed. This is regex and ok for now.  
* The max amount of minutes is set to 59. ~When you set minutes to 59 in your post it will say "one hour", this is the doing of humanize() and will be fixed in a separate issue.~
* Entering a single `0` in any of the fields is fine and that should not be removed.
* The format is not localized yet, this needs to be fixed in a separate issue.

## Quality assurance


Fixes #10663 
Fixes https://github.com/Yoast/wordpress-seo/issues/10668
